lib/AppURLHider.js

```diff
@@ -31,9 +31,13 @@ AppURLHider.prototype = {
   /**
    * Hide the URL of the most recent window.
    */
-  hideURL: function() {
+  _hideURL: function(url) {
     let browserWindow = Services.wm.getMostRecentWindow("navigator:browser");
-    browserWindow.gURLBar.value = "";
+    // Only hide the url if it matches the expected app url.
+    if (browserWindow.gURLBar.value === url) {
```

Comment:
this is checking what the browser chrome is actually displaying. The isAppURL checks the tab actual URL to our list of know app URLs. Make sense? I know, kind of confusing.